From dcbdf7ee4432abc3436f750d971dea5a87dd7512 Mon Sep 17 00:00:00 2001
From: jaishank <jaishankar.rajendran@intel.com>
Date: Mon, 3 Jun 2019 14:40:27 +0530
Subject: [PATCH] Enable generation of Intel(R) AVX/AVX2 Instructions for
 Android NNAPI libraries.

+20-25% Improvement in ML and AI Benchmarks

Test: make vts -j
      run vts run vts-hal run vts-kernel

Change-Id: I01b761be82fee7bdd7976acb9320f0a1457ff70f
---
 nn/Android.bp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/nn/Android.bp b/nn/Android.bp
index 2547164..173cd2a 100644
--- a/nn/Android.bp
+++ b/nn/Android.bp
@@ -22,6 +22,18 @@ cc_defaults {
         "-Werror",
         "-O3",
     ],
+    arch: {
+        x86: {
+            avx2: {
+                cflags: ["-mavx","-mavx2","-mfma"],
+            },
+        },
+        x86_64: {
+            avx2: {
+                cflags: ["-mavx","-mavx2","-mfma"],
+            },
+        },
+    },
     product_variables: {
         debuggable: {  // eng and userdebug builds
             cflags: ["-DNN_DEBUGGABLE"],
-- 
2.7.4

